### ISSUE TITLE
compile fail on windows 10

### ISSUE THREAD

**Issue by aiffc** [JolifantoBambla/vk#9]
## Environment
vk version: vk-20220707-git from quicklisp
CL implementation (name & version): SBCL 2.2.8
OS: windows 10

## How to reproduce the issue
just quickload vk, (ql:quickload :vk)

## Expected behavior
quickload vk success on windows 10

## Actual behavior
quickload vk fail, get an error "The function VK::PROCESS-ARGS is undefined"

## Additional context
error log:
CL-USER> (ql:quickload :vk)
To load "vk":
  Load 1 ASDF system:
    vk
; Loading "vk"
..................................................
..................................................
................

; in:
;      DEFVKFUN (CREATE-INSTANCE VULKAN:CREATE-INSTANCE
;            ((CREATE-INFO (OR INSTANCE-CREATE-INFO CFFI-SYS:FOREIGN-POINTER)))
;            (((ALLOCATOR *DEFAULT-ALLOCATOR*)
;              (OR ALLOCATION-CALLBACKS CFFI-SYS:FOREIGN-POINTER)))
;            :HANDLE-CONSTRUCTOR MAKE-INSTANCE-WRAPPER)
;     (VK::DEFVKFUN (VK:CREATE-INSTANCE VULKAN:CREATE-INSTANCE
;                    ((VK:CREATE-INFO
;                      (OR VK:INSTANCE-CREATE-INFO CFFI-SYS:FOREIGN-POINTER)))
;                    (((VK::ALLOCATOR VK:*DEFAULT-ALLOCATOR*)
;                      (OR VK:ALLOCATION-CALLBACKS CFFI-SYS:FOREIGN-POINTER)))
;                    :HANDLE-CONSTRUCTOR VK:MAKE-INSTANCE-WRAPPER)
;       #<(SIMPLE-ARRAY CHARACTER
;          (682)) Represents [vkCreateInstance](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCreateInstance.html).
;   
;   Args:
;    - CREATE-INFO: a (OR INSTANCE-CREATE-INFO CFFI:FOREIGN-POINTER)
;    - A... {100824533F}>
;       (VK:CREATE-INFO '(:STRUCT VULKAN:INSTANCE-CREATE-INFO) VK:CREATE-INFO :IN)
;       (VK::ALLOCATOR '(:STRUCT VULKAN:ALLOCATION-CALLBACKS) VK::ALLOCATOR :IN
;        :OPTIONAL)
;       (VK:INSTANCE 'VULKAN:INSTANCE VK:INSTANCE :OUT :DISPATCHABLE :HANDLE))
; 
; caught ERROR:
;   (during macroexpansion of (DEFVKFUN (CREATE-INSTANCE VULKAN:CREATE-INSTANCE ...)
;     ...))
;   The function VK::PROCESS-ARGS is undefined.

; in:
;      DEFVKFUN (DESTROY-INSTANCE VULKAN:DESTROY-INSTANCE NIL
;            (((INSTANCE (MAKE-INSTANCE-WRAPPER #)) INSTANCE)
;             ((ALLOCATOR *DEFAULT-ALLOCATOR*)
;              (OR ALLOCATION-CALLBACKS CFFI-SYS:FOREIGN-POINTER)))
;            :TRIVIAL-RETURN-TYPE :TRIVIAL)
;     (VK::DEFVKFUN (VK:DESTROY-INSTANCE VULKAN:DESTROY-INSTANCE NIL
;                    (((VK:INSTANCE #) VK:INSTANCE)
;                     ((VK::ALLOCATOR VK:*DEFAULT-ALLOCATOR*)
;                      (OR VK:ALLOCATION-CALLBACKS CFFI-SYS:FOREIGN-POINTER)))
;                    :TRIVIAL-RETURN-TYPE :TRIVIAL)
;       #<(SIMPLE-ARRAY CHARACTER
;          (357)) Represents [vkDestroyInstance](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkDestroyInstance.html).
;   
;   Args:
;    - INSTANCE (optional): a INSTANCE, defaults to: NIL
;    - ALLOCATOR (... {100829E61F}>
;       (VK:INSTANCE 'VULKAN:INSTANCE VK:INSTANCE :IN :DISPATCHABLE :HANDLE
;        :OPTIONAL)
;       (VK::ALLOCATOR '(:STRUCT VULKAN:ALLOCATION-CALLBACKS) VK::ALLOCATOR :IN
;        :OPTIONAL))
; 
; caught ERROR:
;   (during macroexpansion of (DEFVKFUN (DESTROY-INSTANCE VULKAN:DESTROY-INSTANCE ...)
;     ...))
;   The function VK::PROCESS-ARGS is undefined.

; in:
;      DEFVKFUN (ENUMERATE-PHYSICAL-DEVICES VULKAN:ENUMERATE-PHYSICAL-DEVICES
;            ((INSTANCE INSTANCE)) NIL :HANDLE-CONSTRUCTOR
;            MAKE-PHYSICAL-DEVICE-WRAPPER :COUNT-ARG-NAME PHYSICAL-DEVICE-COUNT
;            :FIRST-ARRAY-ARG-NAME PHYSICAL-DEVICES :ENUMERATE-P T)
;     (VK::DEFVKFUN (VK:ENUMERATE-PHYSICAL-DEVICES
;                    VULKAN:ENUMERATE-PHYSICAL-DEVICES ((VK:INSTANCE VK:INSTANCE))
;                    NIL :HANDLE-CONSTRUCTOR VK:MAKE-PHYSICAL-DEVICE-WRAPPER
;                    :COUNT-ARG-NAME VK:PHYSICAL-DEVICE-COUNT :FIRST-ARRAY-ARG-NAME
;                    VK:PHYSICAL-DEVICES :ENUMERATE-P T)
;       #<(SIMPLE-ARRAY CHARACTER
;          (434)) Represents [vkEnumeratePhysicalDevices](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkEnumeratePhysicalDevices.html).
;   
;   Args:
;    - INSTANCE: a INSTANCE
;   
;   Returns:
;     (CL:VALUES
;     ... {10082B8D2F}>
;       (VK:INSTANCE 'VULKAN:INSTANCE VK:INSTANCE :IN :DISPATCHABLE :HANDLE)
;       (VK:PHYSICAL-DEVICE-COUNT :UINT32 VK:PHYSICAL-DEVICE-COUNT :OUT)
;       (VK:PHYSICAL-DEVICES 'VULKAN:PHYSICAL-DEVICE VK:PHYSICAL-DEVICES :OUT
;                            :DISPATCHABLE :HANDLE :LIST))
; 
; caught ERROR:
;   (during macroexpansion of (DEFVKFUN (ENUMERATE-PHYSICAL-DEVICES VULKAN:ENUMERATE-PHYSICAL-DEVICES ...)
;     ...))
;   The function VK::PROCESS-ARGS is undefined.

and so on

**Comment by JolifantoBambla** [comment]
Thanks for reporting. I'll look into it

**Comment by aiffc** [comment]
I tried to use cmd to load the package then i got this error. I think it should be sbcl‘s bug

here is the error log
C:\Users\admin>sbcl
This is SBCL 2.2.8, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
* (ql:quickload :vk)
To load "vk":
  Load 1 ASDF system:
    vk
; Loading "vk"
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
............................................Heap exhausted during garbage collection: 32 bytes available, 112 requested.
        Immobile Object Counts
 Gen   GF type  fdefn symbol   code  Boxed   Cons    Raw   Code  SmMix  Mixed  LgRaw LgCode  LgMix Waste%       Alloc        Trig   Dirty GCs Mem-age
  2     0    0      0      0      0  12368   1951   2166    247    479    600      0      0     32    0.6   581217808   306949690   17843   1  1.2342
  3     0    0      0      0      0   9761   1674   1310    227    421    480     20      0     78    0.6   454850208     2000000   13971   0  0.0000
  4     0    0      0      0      0      0      0      0      0      0      0      0      0      0    0.0           0     2000000       0   0  0.0000
  5     0    0      0      0      0      0      0      0      0      0      0      0      0      0    0.0           0     2000000       0   0  0.0000
fatal error encountered in SBCL pid 1696416064:
GC invariant lost, file "gencgc.c", line 525

Welcome to LDB, a low-level debugger for the Lisp runtime environment.
(GC in progress, oldspace=2, newspace=3)
ldb>

**Comment by JolifantoBambla** [comment]
Yes, looks like sbcl ran out of memory during compilation. and when you try to load the systems later on, `process-args` is not defined because it's only available during compilation.
My last two weeks were a bit busy, sorry. I'll try to fix it this weekend.

**Comment by aiffc** [comment]
It doesn't matter, thanks. Maybe you could take a rest.

**Comment by Aksej** [comment]
I have the same problem on Debian. Is there a known workaround?

**Comment by JolifantoBambla** [comment]
As far as I know, this only happens during compilation, not at runtime.
So what you could try is to increase the dynamic space (`sbcl --dynamic-space-size <MiB>`) and/or lowering the debug compiler policy (using `restrict-compiler-policy`) when compiling `vk` and using your default settings afterwards.
I still need to figure out a permanent solution.

**Comment by schn4v1d** [comment]
Hi, has there been any update on this? Thanks!
